src/test/java/com/commercetools/project/sync/product/ProductSyncerTest.java

```diff
@@ -15,233 +52,272 @@ void tearDownTest() {
+                assertThat(productDraft.getMasterVariant().getAttributes())
+                    .anySatisfy(
+                        attributeDraft -> {
+                          assertThat(attributeDraft.getName()).isEqualTo("categoryReference");
+                          final JsonNode attributeValue =
+                              AttributeUtils.replaceAttributeValueWithJsonAndReturnValue(
+                                  attributeDraft);
+                          final List<JsonNode> referenceSet =
+                              AttributeUtils.getAttributeReferences(attributeValue);
+                          assertThat(referenceSet)
+                              .anySatisfy(
+                                  reference ->
+                                      assertThat(reference.get("id").asText()).isEqualTo("cat1"));
+                          assertThat(referenceSet)
+                              .anySatisfy(
+                                  reference ->
+                                      assertThat(reference.get("id").asText()).isEqualTo("cat2"));
+                        }));
+
+    assertThat(productDraftKey1)
+        .hasValueSatisfying(
+            productDraft ->
+                assertThat(productDraft.getMasterVariant().getAttributes())
+                    .anySatisfy(
+                        attributeDraft -> {
+                          assertThat(attributeDraft.getName()).isEqualTo("productTypeReference");
+                          final JsonNode attributeValue =
+                              AttributeUtils.replaceAttributeValueWithJsonAndReturnValue(
+                                  attributeDraft);
+                          final List<JsonNode> referenceSet =
+                              AttributeUtils.getAttributeReferences(attributeValue);
+                          assertThat(referenceSet.get(0).get("id").asText()).isEqualTo("prodType1");
+                        }));
+
+    assertThat(testLogger.getAllLoggingEvents()).isEmpty();
+  }
+
+  @Test
+  void transform_WithDiscountedPrices_ShouldRemoveDiscountedPrices() {
+    // preparation
+    final ProjectApiRoot sourceClient = mock(ProjectApiRoot.class);
+    final ProductSyncer productSyncer =
+        ProductSyncer.of(sourceClient, mock(ProjectApiRoot.class), getMockedClock(), null);
+    final ProductProjection productProjection =
+        ProductMixin.toProjection(
+            readObjectFromResource("product-key-10.json", Product.class),
+            ProductProjectionType.STAGED);
+
+    final String jsonStringProductTypes =
+        "{\"data\":{\"productTypes\":{\"results\":[{\"id\":\"53c4a8b4-754f-4b95-b6f2-3e1e70e3d0d3\","
+            + "\"key\":\"prodType1\"}]}}}";
+    final GraphQLResponse graphQLResponse =
+        JsonUtils.fromJsonString(jsonStringProductTypes, GraphQLResponse.class);
+    final ApiHttpResponse apiHttpResponse = mock(ApiHttpResponse.class);
+    when(apiHttpResponse.getBody()).thenReturn(graphQLResponse);
```

Comment:
Please merge from the base branch and use `mockResourceIdsGraphQlRequest()`.

Reply:
[ce26a52](https://github.com/commercetools/commercetools-project-sync/pull/502/commits/ce26a52c1d0141892f818de679abfc4da661b62d)